Libraries/Opc.Ua.Server/Subscription/MonitoredItem.cs

```diff
@@ -1225,8 +1250,11 @@ public virtual bool Publish(OperationContext context, Queue<EventFieldList> noti
                     return false;
                 }
 
-                // only publish if reporting.
-                if (!IsReadyToPublish)
+                // only publish if reporting or resending data
+                ResendDataState selectedToResendData = (ResendDataState)Interlocked.CompareExchange(ref m_resendDataState, (int)ResendDataState.NonResendData, (int)ResendDataState.ResendData);
```

Comment:
Please check if that would also an option: [CompareExchange<T>(T, T, T)](https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.compareexchange?view=netframework-4.6.2#system-threading-interlocked-compareexchange-1(-0@-0-0)), then you could use a bool.

Reply:
This type must be a reference type.